redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,516 @@
+                                        return;
+                                    }
+                                    assemblyData["Model"] = *value;
+                                }
+                            }
+                            },
+                            serviceName, assembly,
+                            "org.freedesktop.DBus.Properties", "GetAll",
+                            "xyz.openbmc_project.Inventory.Decorator."
+                            "Asset");
+                    }
+                    else if (interface == "xyz.openbmc_project.Inventory."
+                                          "Decorator.LocationCode")
+                    {
+                        crow::connections::systemBus->async_method_call(
+                            [aResp, assemblyIndex](
+                                const boost::system::error_code ec3,
+                                const std::variant<std::string>& property) {
+                            if (ec3)
+                            {
+                                BMCWEB_LOG_DEBUG << "DBUS response error";
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+
+                            nlohmann::json& assemblyArray =
+                                aResp->res.jsonValue["Assemblies"];
+                            nlohmann::json& assemblyData =
+                                assemblyArray.at(assemblyIndex);
+
+                            const std::string* value =
+                                std::get_if<std::string>(&property);
+
+                            if (value == nullptr)
+                            {
+                                // illegal value
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            assemblyData["Location"]["PartLocation"]
+                                        ["ServiceLabel"] = *value;
+                            },
+                            serviceName, assembly,
+                            "org.freedesktop.DBus.Properties", "Get",
+                            "xyz.openbmc_project.Inventory.Decorator."
+                            "LocationCode",
+                            "LocationCode");
+                    }
+                }
+            }
+            },
+            "xyz.openbmc_project.ObjectMapper",
+            "/xyz/openbmc_project/object_mapper",
+            "xyz.openbmc_project.ObjectMapper", "GetObject", assembly,
+            std::array<const char*, 5>{
+                "xyz.openbmc_project.Inventory.Item.Vrm",
+                "xyz.openbmc_project.Inventory.Item.Tpm",
+                "xyz.openbmc_project.Inventory.Item.Panel",
+                "xyz.openbmc_project.Inventory.Item.Battery",
+                "xyz.openbmc_project.Inventory.Item.DiskBackplane"});
```

Comment:
It's better if this can be made a variable since we have this same list is used in two places (line# 260-265)